### PR TITLE
PWI improvements

### PR DESCRIPTION
**Pull request by pfrazee**
- Make home and feeds viewable to logged-out users
- Add a signup/signin dialog when attempting to do an action that requires signin

## Mobile

https://github.com/bluesky-social/social-app/assets/1270099/8a70ca6b-b0d1-49f1-a1ad-0b1a0fd6fa73

## Desktop

https://github.com/bluesky-social/social-app/assets/1270099/0a05c260-53a0-47ff-809b-5f5ba1b11509

### CODE REVIEW

File: src/components/dialogs/Signin.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    requestSwitchToAccount({requestedAccount: 'none'})
+  }, [requestSwitchToAccount, closeAllActiveElements])
+
+  const showCreateAccount = React.useCallback(() => {
+    closeAllActiveElements()
+    requestSwitchToAccount({requestedAccount: 'new'})
+  }, [requestSwitchToAccount, closeAllActiveElements])
+
+  return (
+    <Dialog.ScrollableInner
```
Comment (estrattonbailey) [inline src/components/dialogs/Signin.tsx:45]: We could probably make this narrower on web. Prompt does this using `[gtMobile ? {width: 'auto', maxWidth: 400} : a.w_full]`. Would reduce white space on left/right.

Comment (pfrazee) [reply src/components/dialogs/Signin.tsx:45]: Oh nice thanks

Comment (pfrazee) [reply src/components/dialogs/Signin.tsx:45]: Done

File: src/state/queries/feed.ts [hunk 2]
```diff
@@ -216,17 +216,38 @@ const FOLLOWING_FEED_STUB: FeedSourceInfo = {
...
   const {data: preferences, isLoading: isLoadingPrefs} = usePreferencesQuery()
   const pinnedUris = preferences?.feeds?.pinned ?? []
 
   return useQuery({
     staleTime: STALE.INFINITY,
     enabled: !isLoadingPrefs,
-    queryKey: [pinnedFeedInfosQueryKeyRoot, pinnedUris.join(',')],
+    queryKey: [
+      pinnedFeedInfosQueryKeyRoot,
+      hasSession ? 'authed:' : 'unauthed:' + pinnedUris.join(','),
```
Comment (gaearon) [inline src/state/queries/feed.ts]: Precedence oopsie, I think you want `(hasSession ? 'authed:' : 'unauthed:') + pinnedUris.join(','),`

**Comment by gaearon** [comment]
Logging out on web currently shows a big sign in / create screen because the URL is still `/settings`. Should we change that to redirect to `/`?

File: src/state/queries/feed.ts (hunk 2 above)
Comment (gaearon) [reply src/state/queries/feed.ts]: edited on GH

**Review by gaearon — APPROVED** [review]
lgtm

there was a bad query key but i think i fixed it, pls verify

other things are more like nits and not blocking

**Comment by gaearon** [comment]
we can probably show the Feeds tab here now as well

Screenshot 2024-04-12 at 13 28 29

**Comment by gaearon** [comment]
do we need the language toggle? here's the existing screen:

Screenshot 2024-04-12 at 13 36 12

**Pull request merged**